app/Utils/SubmissionUtils.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Utils;
+
+class SubmissionUtils
+{
+
+    /**
+     * Figure out what type of XML file this is
+     */
+    public static function get_xml_type(mixed $filehandle): mixed
```

Comment:
Is there a reason why accepting a file handle is more optimal than just passing a file path?

Reply:
No good reason, I was trying to make the most minimum code changes with respect to the old behavior of the `ctest_parse()` function in `ctestparser.php`. But looking at it again now, we should simplify this.

Reply:
Looking at this again, it looks like the parent function to `get_xml_type()` in the parsing code is `ctest_parse()`, which itself doesn't know the file path. Ideally we should implement your suggestion, but it would require more code infrastructure changes than I think is appropriate for this PR. We should follow up on this PR with the improvements/refacors in a future PR.